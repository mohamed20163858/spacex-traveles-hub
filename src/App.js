import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import Categories from './components/Categories';
import Books from './components/Books';
import Navbar from './components/Navbar';

function App() {
  return (
    <Router>
      <Navbar />
      <Routes>
        <Route path="/Categories" element={<Categories />} />
        <Route path="/" element={<Books />} />
      </Routes>
    </Router>
  );
}

export default App;
