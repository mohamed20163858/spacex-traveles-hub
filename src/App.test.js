import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import store from './redux/configureStore';
import '@testing-library/jest-dom';
import App from './App';

function TestApp() {
  return (
    <Provider store={store}>
      <App />
    </Provider>
  );
}
it('renders correctly', () => {
  const tree = renderer
    .create(<TestApp />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('simulate navigating  through Rockets link ', () => {
  // ARRANGE
  render(<TestApp />);

  // ACT
  userEvent.click(screen.getByText('Rockets'));
  // Arrange
  const result = screen.getByTestId('momo-id');
  // ASSERT
  expect(result).toBeInTheDocument();
});

test('simulate navigating  through Missions link ', () => {
  // ARRANGE
  render(<TestApp />);

  // ACT
  userEvent.click(screen.getByText('Missions'));
  // Arrange
  const result = screen.getByTestId('momo-id');
  // ASSERT
  expect(result).toBeInTheDocument();
});
test('simulate navigating  through My Profile link ', () => {
  // ARRANGE
  render(<TestApp />);

  // ACT
  userEvent.click(screen.getByText('My Profile'));
  // Arrange
  const result = screen.getByTestId('momo-id');
  // ASSERT
  expect(result).toBeInTheDocument();
});
