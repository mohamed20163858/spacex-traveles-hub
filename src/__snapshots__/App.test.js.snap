// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="outer"
>
  <div
    className="inner-1"
  >
    <a
      className="logo"
      href="/my-profile"
      onClick={[Function]}
    >
      <img
        alt="logo"
        height="40"
        src="logo.png"
        width="40"
      />
      <p>
        Space Travelers'Hub
      </p>
    </a>
  </div>
  <div
    className="inner-2"
  >
    <ul>
      <li>
        <a
          className=""
          href="/rockets"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className=""
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <div
        className="vertical-line"
      />
      <li>
        <a
          className=""
          href="/my-profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </div>
</div>
`;
